js/views/callinfoview.js

```diff
@@ -64,75 +60,128 @@
 		renderTimeout: undefined,
 
 		templateContext: function() {
-			var canModerate = this.model.get('participantType') === 1 || this.model.get('participantType') === 2;
+			var canModerate = this._canModerate();
 			return $.extend(this.model.toJSON(), {
+				isGuest: this.model.get('participantType') === 4,
 				canModerate: canModerate,
 				isPublic: this.model.get('type') === 3,
 				showShareLink: !canModerate && this.model.get('type') === 3,
-				isNameEditable: canModerate && this.model.get('type') !== 1,
 				isDeletable: canModerate && (Object.keys(this.model.get('participants')).length > 2 || this.model.get('numGuests') > 0)
 			});
 		},
 
 		ui: {
-			'roomName': 'h3.room-name',
+			'roomName': 'div.room-name',
 			'clipboardButton': '.clipboard-button',
 			'linkCheckbox': '.link-checkbox',
 
-			'renameButton': '.rename-button',
-			'renameOption': '.rename-option',
-			'renameInput': '.rename-input',
-			'renameConfirm': '.rename-confirm',
+			'guestName': 'div.guest-name',
 
 			'passwordOption': '.password-option',
 			'passwordInput': '.password-input',
 			'passwordConfirm': '.password-confirm'
 		},
 
+		regions: {
+			'roomName': '@ui.roomName',
+			'guestName': '@ui.guestName'
+		},
+
 		events: {
 			'change @ui.linkCheckbox': 'toggleLinkCheckbox',
 
-			'click @ui.renameButton': 'showRenameInput',
-			'keyup @ui.renameInput': 'keyUpRename',
-			'click @ui.renameConfirm': 'confirmRename',
-
 			'keyup @ui.passwordInput': 'keyUpPassword',
 			'click @ui.passwordConfirm': 'confirmPassword'
 		},
 
 		modelEvents: {
-			'change:displayName': function() {
-				this.renderWhenInactive();
-			},
 			'change:hasPassword': function() {
 				this.renderWhenInactive();
 			},
 			'change:participantType': function() {
+				this._updateNameEditability();
+
 				// User permission change, refresh even when typing, because the
 				// action will fail in the future anyway.
 				this.render();
 			},
 			'change:type': function() {
+				this._updateNameEditability();
+
 				this.renderWhenInactive();
 			}
 		},
 
+		initialize: function() {
+			this._nameEditableTextLabel = new OCA.SpreedMe.Views.EditableTextLabel({
+				model: this.model,
+				modelAttribute: 'displayName',
+				modelSaveOptions: {
+					patch: true,
+					success: function() {
+						// Renaming a room by setting "displayName" causes "name" to
+						// change too in the server, so the model has to be fetched
+						// again to get the changes.
+						this.model.fetch();
+					}
+				},
+
+				extraClassNames: 'room-name',
+				labelTagName: 'h3',
+				inputMaxLength: '200',
+				inputPlaceholder: t('spreed', 'Name'),
+				buttonTitle: t('spreed', 'Rename')
+			});
+
+			this._updateNameEditability();
+
+			this._guestNameEditableTextLabel = new OCA.SpreedMe.Views.EditableTextLabel({
+				model: this.getOption('guestNameModel'),
+				modelAttribute: 'nick',
+
+				extraClassNames: 'guest-name',
+				labelTagName: 'p',
+				labelPlaceholder: t('spreed', 'Guest'),
```

Comment:
Should we use something like "Set your name" here? Making it more pressuring to set a name?

Reply:
Or can we add a label next to it?

Especially when the room has no name set, you see "Admin, You" and next line "Guest". This is a bit confusing.

Reply:
Yes, I agree that just the label there with _Guest_ is terrible.

@jancborchardt Any advice on this? Should I add another label next to it like _Your name:_ or simply show _Set your name_ in the current label when there is no name set? Any other alternative?

Reply:
Yeah, the placeholder should be »Your name …«

Reply:
Thanks, done!